### ISSUE TITLE
Transfer learning with pretrained weights doesn't work

### ISSUE THREAD

**Issue by dinarkino** [hustvl/YOLOP#123]
I did several experiments and faced the situation when the model couldn't learn. The first time I faced the problem, when I trained the model on a different dataset. Then I decided to debug this on just one image for simplicity reasons. I used ENC_DRIVABLE_ONLY flag with such a setup for training from scratch (and DRIVABLE_ONLY for training from pretrained weights):

        if cfg.TRAIN.ENC_DRIVABLE_ONLY:  # Only train encoder and DA area segmentation branch
            logger.info('freeze Det and Ll_Seg heads...')
            for k, v in model.named_parameters():
                v.requires_grad = True  # train all layers 
                if k.split(".")[1] in Det_Head_para_idx + Ll_Seg_Head_para_idx:
                    print('freezing %s' % k)
                    v.requires_grad = False

There is a road in the image which I wanted to segment. In the case of training without pretrained weights, after some amount of epochs (around 70), model starts to train and metrics increase steadily until 85% of IoU. At the same time, when I do the same with pretrained weights, I see that the model starts at 35% of IoU, then reaches around 50% and then forgets everything and metrics steadily go to 10-20% of IoU. This behavior is very strange. I wonder if anyone faced the same? 

Also, during debugging I noticed that when we load pretrained weights there is also load of optimizer's state_dict: 

https://github.com/hustvl/YOLOP/blob/084d0bb19fa2db4ba40a6e1d6b120f0153d2f335/tools/train.py#L158

It overrides optimizer's parameters from the config, which could lead to misleading results. After deletion of this line, I could make model train even with pretrained weights, but only for large lr (like 0.1 for adam). For a smaller lr the behavior is the same as before.

**Comment by dinarkino** [comment]
The problem was in my data preparation. I didn't notice that authors do `self.Tensor = transforms.ToTensor()` for segmentation masks in range 0-255. I did the same for one-hot masks and though that we just created tensors without transform. 

After this fix the model trains well on new data with pre-trained weights.
